Problems with building guile version 1.8.8 on Solaris SPARC:

.../components/guile/guile-1.8.8/libguile/gc_os_dep.c:720:37: error: 'USERLIMIT' undeclared (first use in this function)
 #       define STACKBOTTOM ((ptr_t) USRSTACK)
...

See guile-devel email thread at:

  https://lists.gnu.org/archive/html/guile-devel/2011-04/msg00236.html

for more details.

Upstream already know about this.

--- guile-1.8.8/libguile/gc_os_dep.c.orig	Mon Dec 13 19:25:01 2010
+++ guile-1.8.8/libguile/gc_os_dep.c	Fri Apr 15 14:03:13 2011
@@ -714,11 +714,8 @@
 /*      # define STACKBOTTOM ((ptr_t)(_start)) worked through 2.7,      */
 /*      but reportedly breaks under 2.8.  It appears that the stack     */
 /*      base is a property of the executable, so this should not break  */
 /*      old executables.                                                */
-/*      HEURISTIC2 probably works, but this appears to be preferable.   */
-#       include <sys/vm.h>
-#       define STACKBOTTOM ((ptr_t) USRSTACK)
 #	ifndef USE_MMAP
 #	    define USE_MMAP
 #	endif
 #       ifdef USE_MMAP
